Cython/Compiler/MemoryView.py

```diff
@@ -266,9 +281,18 @@ def generate_buffer_slice_code(self, code, indices, dst, dst_type, have_gil,
         """
         src = self.cname
 
+        # There's a decent chance that the source and destination memoryviews are the same
+        # in which case we can skip the reference counting for speed
+        code.putln("if (%(dst)s.memview != %(src)s.memview) {" % locals())
+        code.put_xdecref(dst, self.type, have_gil=have_gil)
         code.putln("%(dst)s.data = %(src)s.data;" % locals())
         code.putln("%(dst)s.memview = %(src)s.memview;" % locals())
         code.put_incref_memoryviewslice(dst, dst_type, have_gil=have_gil)
+        code.putln("} else {")
+        # just copy the data
+        code.putln("%(dst)s.data = %(src)s.data;" % locals())
+        code.putln("}")
```

Comment:
Is it really worth the complexity of passing the `locals()` here and doing dict string formatting? I'm pretty sure we do not do that anywhere else, and it's really just two simple variables.

Reply:
No - it's just a style copied from the surrounding code. But I'll replace it next iteration